src/MemoryTracing/MemoryTracingIntegrationTest.cpp

```diff
@@ -131,120 +152,76 @@ class MemoryTracingIntegrationTestFixture {
-        EXPECT_TRUE(event.process_memory_usage().rss_anon_kb() >= 0);
-        EXPECT_TRUE(event.process_memory_usage().minflt() >= 0);
-        EXPECT_TRUE(event.process_memory_usage().majflt() >= 0);
-
-        previous_process_memory_event_timestamp_ns = current_process_memory_event_timestamp_ns;
-        break;
-      }
-      default:
-        UNREACHABLE();
+    EXPECT_TRUE(event.event_case() == ProducerCaptureEvent::kMemoryUsageEvent);
+    const MemoryUsageEvent& memory_usage_event = event.memory_usage_event();
+
+    // Verify whether events are in order of their timestamps.
+    uint64_t current_memory_usage_event_timestamp_ns = memory_usage_event.timestamp_ns();
+    EXPECT_GE(current_memory_usage_event_timestamp_ns, previous_memory_usage_event_timestamp_ns);
+
+    std::vector<uint64_t> timestamps;
+    // Verify whether the contents of system memory events are valid.
+    EXPECT_TRUE(memory_usage_event.has_system_memory_usage());
+    const SystemMemoryUsage& system_memory_usage = memory_usage_event.system_memory_usage();
+    timestamps.push_back(system_memory_usage.timestamp_ns());
+    EXPECT_TRUE(system_memory_usage.total_kb() >= 0);
+    EXPECT_TRUE(system_memory_usage.free_kb() >= 0);
+    EXPECT_TRUE(system_memory_usage.available_kb() >= 0);
+    EXPECT_TRUE(system_memory_usage.buffers_kb() >= 0);
+    EXPECT_TRUE(system_memory_usage.cached_kb() >= 0);
+    EXPECT_TRUE(system_memory_usage.pgfault() >= 0);
+    EXPECT_TRUE(system_memory_usage.pgmajfault() >= 0);
+
+    // Verify whether the contents of process memory events are valid.
+    EXPECT_TRUE(memory_usage_event.has_process_memory_usage());
+    const ProcessMemoryUsage& process_memory_usage = memory_usage_event.process_memory_usage();
+    timestamps.push_back(process_memory_usage.timestamp_ns());
+    EXPECT_TRUE(process_memory_usage.rss_anon_kb() >= 0);
+    EXPECT_TRUE(process_memory_usage.minflt() >= 0);
+    EXPECT_TRUE(process_memory_usage.majflt() >= 0);
+
+    // If cgroup memory events are collected, verify whether the contents are valid.
+    if (memory_usage_event.has_cgroup_memory_usage()) {
+      const CGroupMemoryUsage& cgroup_memory_usage = memory_usage_event.cgroup_memory_usage();
+      EXPECT_FALSE(cgroup_memory_usage.cgroup_name().empty());
+      timestamps.push_back(cgroup_memory_usage.timestamp_ns());
+      EXPECT_TRUE(cgroup_memory_usage.limit_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.rss_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.mapped_file_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.pgfault() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.pgmajfault() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.unevictable_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.inactive_anon_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.active_anon_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.inactive_file_bytes() >= 0);
+      EXPECT_TRUE(cgroup_memory_usage.active_file_bytes() >= 0);
```

Comment:
Prefer `EXPECT_GE` to `EXPECT_TRUE(... > ...)`

Reply:
Done. Thanks!